lib/controller/LocationController.js

```diff
@@ -203,20 +205,21 @@ LocationController.push({
   handler: async (request) => {
     const { s1 } = request.query;
     const features = CompositeId.decode(s1);
-    if (features.length > MAX_LOCATIONS) {
-      return Boom.badRequest(`cannot route corridor on more than ${MAX_LOCATIONS} locations`);
+    const featuresSelection = {
+      features,
+      selectionType: LocationSelectionType.CORRIDOR,
+    };
+    try {
+      const featuresResolved = await FeatureResolver.byFeaturesSelection(featuresSelection);
+      const locations = await CentrelineDAO.byFeatures(featuresResolved);
+      return filterLocations(locations);
+    } catch (err) {
+      if (err instanceof InvalidFeaturesSelectionError) {
+        const { statusCode } = HttpStatus.BAD_REQUEST;
```

Comment:
as we found with the test, this _does_ mean that "corridor not found" results in HTTP 400, not HTTP 404.  (this isn't urgent or even necessary to fix, especially while we don't have any status code-specific handling in the frontend!)